tests/e2e/ans/ans.test.ts

```diff
@@ -0,0 +1,175 @@
+  describe("isValidANSName", () => {
+    test("it returns true for valid names", () => {
+      expect(isValidANSName("primary")).toEqual({ domainName: "primary", subdomainName: undefined });
+      expect(isValidANSName("primary.apt")).toEqual({ domainName: "primary", subdomainName: undefined });
+
+      expect(isValidANSName("secondary.primary")).toEqual({ domainName: "primary", subdomainName: "secondary" });
+      expect(isValidANSName("secondary.primary.apt")).toEqual({ domainName: "primary", subdomainName: "secondary" });
+
+      expect(isValidANSName({ domainName: "primary", subdomainName: "secondary" })).toEqual({
+        domainName: "primary",
+        subdomainName: "secondary",
+      });
+    });
+
+    test("it returns false for invalid names", () => {
+      expect(() => isValidANSName("")).toThrow();
+      expect(() => isValidANSName(".")).toThrow();
+      expect(() => isValidANSName("..")).toThrow();
+      expect(() => isValidANSName(" . ")).toThrow();
+      expect(() => isValidANSName(" test ")).toThrow();
+      expect(() => isValidANSName(".apt")).toThrow();
+      expect(() => isValidANSName(".apt.apt")).toThrow();
+      expect(() => isValidANSName(".apt.")).toThrow();
+      expect(() => isValidANSName("1")).toThrow();
+      expect(() => isValidANSName("1.apt")).toThrow();
+      expect(() => isValidANSName("bad.bad.bad")).toThrow();
+      expect(() => isValidANSName({ domainName: "1" })).toThrow();
+      expect(() => isValidANSName({ domainName: "1", subdomainName: "2" })).toThrow();
+    });
+  });
+
+  describe("registerName", () => {
+    test("can be called with a variety of parameters", async () => {
+      const alice = Account.generate();
+      await aptos.fundAccount({
+        accountAddress: alice.accountAddress.toString(),
+        amount: 500_000_000,
+      });
+      const name = randomString();
+
+      expect(
+        await aptos.ans.registerName({
+          name,
+          sender: alice,
+          expiration: { policy: "domain", years: 1 },
+        }),
+      ).toBeTruthy();
+
+      await expect(
+        aptos.ans.registerName({
+          sender: alice,
+          name,
+          // Force the year to be absent
+          expiration: { policy: "domain" } as any,
+        }),
+      ).rejects.toThrow();
+
+      // Testing to make sure that the subdomain policy is enforced
+      await expect(
+        aptos.ans.registerName({
+          sender: alice,
+          name,
+          // Force the year to be absent
+          expiration: { policy: "subdomain:follow-domain" },
+        }),
+      ).rejects.toThrow();
+    });
+
+    test("it mints a domain name and gives it to the sender", async () => {
+      const name = randomString();
+      const alice = Account.generate();
+      await aptos.fundAccount({
+        accountAddress: alice.accountAddress.toString(),
+        amount: 500_000_000,
+      });
+
+      await signAndSubmit(
+        alice,
+        await aptos.ans.registerName({
```

Comment:
```suggestion
        await aptos.registerName({
```

Reply:
Any particular reason to drop the namespace in this particular instance (and the one bellow)?

Reply:
no particular reason with those specific cases, it is in general that we can drop the namespace - same as we do with everything. Just didn't want to spam the PR with the same comment everywhere